components/list/list-item.js

```diff
@@ -125,200 +40,52 @@ class ListItem extends RtlMixin(LitElement) {
-		this.requestUpdate('breakpoints', oldValue);
-	}
-
-	connectedCallback() {
-		super.connectedCallback();
-
-		const separators = this.parentNode.getAttribute('separators');
-		if (separators) this._separators = separators;
-		this._extendSeparators = this.parentNode.hasAttribute('extend-separators');
-
-		ro.observe(this);
-	}
-
-	disconnectedCallback() {
-		super.disconnectedCallback();
-		ro.unobserve(this);
-	}
-
-	focus() {
-		const node = getFirstFocusableDescendant(this);
-		if (node) node.focus();
-	}
-
-	render() {
-
-		const label = this.selectable ? html`<label class="d2l-list-item-label" for="${this._checkBoxId}" aria-labelledby="${this._contentId}"></label>` : null;
-		const link = this.href ? html`<a class="d2l-list-item-link" href="${ifDefined(this.href)}" aria-labelledby="${this._contentId}"></a>` : null;
-		const beforeContent = this.selectable
-			? html`<input id="${this._checkBoxId}" class="d2l-input-checkbox" @change="${this._handleCheckboxChange}" type="checkbox" .checked="${this.selected}"><slot name="illustration"></slot>`
-			: html`<slot name="illustration"></slot>`;
-
-		const classes = {
-			'd2l-list-item-container': true,
-			'd2l-list-item-flex': label || link || this.illustrationOutside,
-			'd2l-visible-on-ancestor-target': true
-		};
-
-		return html`
-			<div class="${classMap(classes)}" breakpoint="${this._breakpoint}">
-				${label}
-				${this.illustrationOutside ? beforeContent : null}
-				${link}
-				<div id="${this._contentId}"
-					class="d2l-list-item-content"
-					?extend-separators="${this._extendSeparators}"
-					separators="${ifDefined(this._separators)}">
-					<div class="d2l-list-item-content-flex">
-						${!this.illustrationOutside ? beforeContent : null}
-						<div class="d2l-list-item-main"><slot></slot></div>
-						<slot name="actions"></slot>
-					</div>
-				</div>
-			</div>
-		`;
-
-	}
-
-	resizedCallback(width) {
-		const lastBreakpointIndexToCheck = 3;
-		this.breakpoints.some((breakpoint, index) => {
-			if (width >= breakpoint || index > lastBreakpointIndexToCheck) {
-				this._breakpoint = lastBreakpointIndexToCheck - index - (lastBreakpointIndexToCheck - this.breakpoints.length + 1) * index;
-				return true;
-			}
-		});
-	}
-
-	setSelected(selected, suppressEvent) {
-		this.selected = selected;
-		if (!suppressEvent) this._dispatchSelected(selected);
-	}
-
-	updated(changedProperties) {
-		if (changedProperties.has('key')) {
-			const oldValue = changedProperties.get('key');
-			if (typeof oldValue !== 'undefined') {
-				this.setSelected(undefined, true);
-			}
-		}
-		if (changedProperties.has('breakpoints')) {
-			this.resizedCallback(this.offsetWidth);
-		}
+	_renderIllustration() {
+		return html`<slot name="illustration"></slot>`;
 	}
 
-	_dispatchSelected(value) {
-		this.dispatchEvent(new CustomEvent('d2l-list-item-selected', {
-			detail: { key: this.key, selected: value },
-			bubbles: true
-		}));
+	_renderOutside() {
```

Comment:
By redefining these methods we can add elements in the more complicated `render()` method in the mixin. Makes things simple.